torchrl/envs/common.py

```diff
@@ -1816,25 +1838,109 @@ def policy(td):
             )
             # done and truncated are in done_keys
             # We read if any key is done.
-            any_done = terminated_or_truncated(
+            any_done = _terminated_or_truncated(
                 tensordict,
                 full_done_spec=self.output_spec["full_done_spec"],
-                key=None if break_when_any_done else "_reset",
+                key=None,
             )
-            if break_when_any_done and any_done:
+            if any_done:
                 break
-            if not break_when_any_done and any_done:
-                tensordict = self.reset(tensordict)
 
             if callback is not None:
                 callback(self, tensordict)
+        return tensordicts
 
-        batch_size = self.batch_size if tensordict is None else tensordict.batch_size
-        out_td = torch.stack(tensordicts, len(batch_size))
-        if return_contiguous:
-            out_td = out_td.contiguous()
-        out_td.refine_names(..., "time")
-        return out_td
+    def _rollout_nonstop(
+        self,
+        *,
+        tensordict,
+        auto_cast_to_device,
+        max_steps,
+        policy,
+        policy_device,
+        env_device,
+        callback,
+    ):
+        tensordicts = []
+        tensordict_ = tensordict
+        for i in range(max_steps):
+            if auto_cast_to_device:
+                tensordict_ = tensordict_.to(policy_device, non_blocking=True)
+            tensordict_ = policy(tensordict_)
+            if auto_cast_to_device:
+                tensordict_ = tensordict.to(env_device, non_blocking=True)
```

Comment:
```suggestion
                tensordict_ = tensordict_.to(env_device, non_blocking=True)
```
I want a million candies.

Also how did this pass our tests? maybe let's test this